apps/amiapp_flutter/pubspec.lock

```diff
@@ -678,4 +678,4 @@ packages:
     version: "3.1.2"
 sdks:
   dart: ">=3.3.0 <4.0.0"
-  flutter: ">=3.18.0-18.0.pre.54"
+  flutter: ">=3.7.0"
```

Comment:
I see this and a few other changes for downgrading versions. Could this be the reason for getting those errors? Can you try updating Flutter version and see if that helps?

If that fixes the problem, would be happy to discuss internally on how can we make this consistent across our team and have clear recommendations for versions. But I want to make sure we understand the root cause and we have a reason to downgrade these versions.

Reply:
Oh, that is weird. I am not sure why this happened? 

Let me try an upgrade first and see what happens.

Reply:
Not sure what caused these lock file changes but I seem to have fixed it. 

The iOS app is compiling successfully with the code in this PR. Ready for re-review.